subsys/usb/device_next/class/usbd_cdc_ncm.c

```diff
@@ -0,0 +1,1278 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * Copyright (c) 2024 Hardy Griech
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Specification
+ * -------------
+ * NCM spec can be obtained here:
+ * https://www.usb.org/document-library/network-control-model-devices-specification-v10-and-errata-and-adopters-agreement
+ *
+ * Small Glossary (from the spec)
+ * --------------
+ * Datagram - A collection of bytes forming a single item of information, passed as a unit
+ *            from source to destination.
+ * NCM      - Network Control Model
+ * NDP      - NCM Datagram Pointer: NTB structure that delineates Datagrams
+ *            (typically Ethernet frames) within an NTB
+ * NTB      - NCM Transfer Block: a data structure for efficient USB encapsulation of
+ *            one or more datagrams.  Each NTB is designed to be a single USB transfer
+ * NTH      - NTB Header: a data structure at the front of each NTB, which provides the
+ *            information needed to validate the NTB and begin decoding
+ *
+ * Some explanations
+ * -----------------
+ * - itf_data_alt  if != 0 -> data xmit/recv are allowed (see spec)
+ * - ep_in         IN endpoints take data from the device intended to go in to the host
+ *                 (the device transmits)
+ * - ep_out        OUT endpoints send data out of the host to the device (the device receives)
+ *
+ * Linux host NCM driver
+ * ---------------------
+ * - https://github.com/torvalds/linux/blob/master/drivers/net/usb/cdc_ncm.c
+ * - https://github.com/torvalds/linux/blob/master/include/linux/usb/cdc_ncm.h
+ * - https://github.com/torvalds/linux/blob/master/include/uapi/linux/usb/cdc.h
+ */
+
+#define DT_DRV_COMPAT zephyr_cdc_ncm_ethernet
+
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/ethernet.h>
+
+#include <eth.h>
```

Comment:
Is this `<zephyr/drivers/ethernet/eth.h>`? Was there a reason for leaving-off the prefix? Just curious.

Reply:
not sure, but again... taken directly from the ECM driver :-)